src/platform_impl/macos/window.rs

```diff
@@ -575,6 +575,14 @@ impl UnownedWindow {
     }
   }
 
+  #[inline]
+  pub fn is_focused(&self) -> bool {
+    unsafe {
+      let is_key_window: BOOL = msg_send![*self.ns_window, isKeyWindow];
+      is_key_window == YES
+    }
+  }
```

Comment:
I am not sure if this enough, `Window::set_focus` calls `makeKeyAndOrderFront` and you are only checking for the key part, can we also check for the front order? I am not a macOS dev so I will let @wusyong approve this one

Reply:
Will probably need to check the order too. But I fear it will become a little bit too difficult. It's good to merge until we see any edge case I guess.

Reply:
@henry40408 , do you think you can check for order too in this PR? or should I merge?

Reply:
@amrbashir I'd like to try checking order as well. Please do not merge.

Reply:
According to [Window Types and Behavior](https://www.oreilly.com/library/view/building-cocoa-applications/0596002351/ch01s05.html), I should use [`isMainWindow`](https://developer.apple.com/documentation/appkit/nswindow/1419130-ismainwindow) instead. I copy the paragraph here in case the source is missing:

> The key window is the window or dialog that will respond to the Mac keyboard. The main window is the document window that corresponds to the **active document** (e.g., a document window in a word processor or image window in a graphics editor).

I try to search for definitions of key window and main window in Apple documentation but no luck, so I provided alternative source here to clarify why I use isMainWindow instead.

Reply:
from the link you provided for `isMainWindow`:
> The value of this property is YES when the window is the main window for the application; otherwise, NO.

So `isMainWindow` will only return true if it is the main window of the application, so I don't think that's it. 

Also we should keep the check for `keyWindow`, we need to have two checks:
1. is this window a `keyWindow`?
2. is this key the active window (probably the top most foreground window)?

Reply:
I modified `is_focused` so it will only return true when the following conditions are true:

1. is this window a key window ➡️ `isKeyWindow`
2. is this key the active window ➡️ `isMainWindow`

Reply:
we can't depend on `isMainWindow` because it doesn't take other applications in consideration and it is not necessary that the mainWindow is the active window

Reply:
So we can just check `isKeyWindow` only?

Reply:
we also want to check if this is top most window, if there is no API for that, we can just isKeyWindow only

Reply:
I guess we can use `isKeyWindow` only for now, thanks for looking into it